freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -96,63 +135,144 @@ def handle(self, event):
         # by the highest semantic version
         all_bundles = self._pyxis.get_latest_bundles(index_images)
 
-        bundles = self._pyxis.filter_bundles_by_related_image_digests(
-            original_images_digests, all_bundles)
+        # get bundle digests for original images
+        bundle_digests_by_related_nvr = {}
+        for image_nvr, image_digest in original_digests_by_nvr.items():
+            bundles = self._pyxis.get_bundles_by_related_image_digest(
+                image_digest, all_bundles
+            )
+            if not bundles:
+                log.info(f"No latest bundle image found for {image_nvr}")
+                continue
+
+            for bundle in bundles:
+                bundle_digest = bundle['bundle_path_digest']
+                bundle_digests_by_related_nvr.setdefault(image_nvr, []).append(bundle_digest)
+
+        if not bundle_digests_by_related_nvr:
+            msg = "None of the original images have related bundles, skip."
+            log.warning(msg)
+            db_event.transition(EventState.SKIPPED, msg)
+            return []
+
+        # A set of unique bundle digests, an image can be in multiple bundles' related_images
         bundle_digests = set()
-        for bundle in bundles:
-            if not bundle.get('bundle_path_digest'):
-                log.warning("Bundle %s doesn't have 'bundle_path_digests' set",
-                            bundle['bundle_path'])
+        for digests in bundle_digests_by_related_nvr.values():
+            bundle_digests.update(digests)
```

Comment:
Optional: You could populate this set at the same time that you populate `bundle_digests_by_related_nvr`. It avoid an additional loop over this dictionary and it makes it easier to read.